src/plugins/intel_gpu/tests/unit/passes/prepare_buffer_fusing_test.cpp

```diff
@@ -363,3 +363,70 @@ TEST(prepare_buffer_fusing, crop_b_axis) {
         }
     }
 }
+
+#ifdef ENABLE_ONEDNN_FOR_GPU
+TEST(prepare_buffer_fusing, in_place_onednn_concat_static) {
+    auto& engine = get_test_engine();
+
+    if (!engine.get_device_info().supports_immad)
+       return;
+
+    auto in_layout1 = layout{ ov::PartialShape{1, 1, 4, 2}, data_types::f32, format::bfyx };
+    auto in_layout2 = layout{ ov::PartialShape{1, 2, 4, 2}, data_types::f32, format::bfyx };
+
+    topology topology;
+    topology.add(input_layout("input1", in_layout1));
+    topology.add(input_layout("input2", in_layout2));
+    topology.add(reorder("reorder1", input_info("input1"), format::bfyx, data_types::f16));
+    topology.add(reorder("reorder2", input_info("input2"), format::bfyx, data_types::f16));
+    topology.add(concatenation("concat", { input_info("reorder1"), input_info("reorder2") }, 1));
+    topology.add(reorder("output", input_info("concat"), format::bfyx, data_types::f32));
+
+    ExecutionConfig config = get_test_default_config(engine);
+    config.set_property(ov::intel_gpu::optimize_data(true));
+    config.set_property(ov::intel_gpu::allow_new_shape_infer(false));
+    ov::intel_gpu::ImplementationDesc impl = { format::bfyx, std::string(""), impl_types::onednn };
+    config.set_property(ov::intel_gpu::force_implementations(ov::intel_gpu::ImplForcingMap{ {"concat", impl} }));
```

Comment:
I think this concat is supposed to be optimized out. Why is it necessary to force impl?

Reply:
Just cp from other test, I'll update to remove it.